### ISSUE TITLE
Documentation Suggestion

### ISSUE THREAD

**Issue by Patrick-Seymour** [andrewgu/ModConfigMenu#28]
tldr: ScreenClass=class'MCM_OptionsScreen' compiles to None if only compiling API without main codebase, so at the very least a note about this should be in the documentation, ideally in long form paragraph where appropriate and short form comment in example code.

This is just a suggestion for some information to add to your documentation.

Basically, when I was building MCM straight in with my own mod, everything worked fine for the options menu listener. But when I removed the MCM main code (keeping the API stuff), I realized that the listener was starting to listen to everything instead of just the MCM menu.

```
defaultproperties
{
  ScreenClass=class'MCM_OptionsScreen'
}
```

Basically, because the compiler can't actually find the class, it just prints a WARNING (not an ERROR) to the compile log file and successfully compiles the project, but in actuality it replaces all class instances with NONE, so the above is equivalent to "ScreenClass=none".

That one other Issue that you dealt with recently of MCM's own ScreenClass listener, reminded me of this. Basically there are two ways to get this working for a mod listening for MCM.

The first is to just listen to ScreenClass=none and do the UIScreen instance testing against the MCM_API interface only. This does mean that the test is done on every UIScreen instance, but it is the most accurate and isn't too much of a hit.

```
local MCM_API APIScreen;
APIScreen = MCM_API(Screen);
if (APIScreen != none) {
  ...
}
```

The second is to use one of the following to retrieve either the MCM_OptionsScreen class or the class of a mod replacement for it, should there be a mod replacing it:

```
local class<UIScreen> MCMOptionsScreenClassType;
local class<UIScreen> MCMOptionsScreenModOverride;
if (ScreenClass == none) {
  MCMOptionsScreenClassType = class'XComGame'.static.FindClassType("MCM_OptionsScreen");
  MCMOptionsScreenModOverride = `XENGINE.GetModReplacementClass(MCMOptionsScreenClassType);
  ScreenClass = ...;
}
// still do the screen test
```

**Comment by andrewgu** [comment]
Oh I see, yeah, ok, that definitely needs fixing. Good catch!

**Comment by andrewgu** [comment]
I'm going to put this in the 1.1 release since I believe it may affect the API macros.

**Comment by andrewgu** [comment]
Turns out an API change isn't needed. I took your second suggestion in the documentation and updated the example code to reflect as well.

Thanks for catching this issue!
